### ISSUE TITLE
Teamviewer uninstall bricks El Capitain

### ISSUE THREAD

**Issue by guandalf** [caskroom/homebrew-cask#15322]
***WARNING***

Uninstalling teamviewer with "brew uninstall teamviewer" renders El Capitain unusable. The system boots up but no login screen is shown. I had to erase and reinstall (I had no time machine available).

Very dangerous.

**Comment by vitorgalvao** [comment]
It’s highly unlikely that was the issue, particularly since you have given no details on what you were doing other than that.

However, if this is the case, it is indeed serious, so I’m testing it on a clean VM right now.

**Comment by adityadalal924** [comment]
The uninstall stanza in question:

```
  uninstall :pkgutil   => 'com.teamviewer.*',
            :launchctl => 'com.teamviewer.service',
            :delete    => '/Library/LaunchDaemons/com.teamviewer.teamviewer_service.plist'
```

While I am on El Capitan and have installed and uninstalled Teamviewer, I haven't had any bricking problems. I don't have a clean VM to test with at the moment so waiting on @vitorgalvao's results.

**Comment by vitorgalvao** [comment]
Done. Everything worked as expected. Your problem lied elsewhere, @guandalf, and without further details there isn’t anything we can do to diagnose what caused it, but it seems clear it wasn’t caused by us.

On closer inspection of your comment, though, you wrote `brew uninstall teamviewer`, with `cask` missing. Either that was a mistake (in which case we’ve already covered it), or if you really had a formula that allowed you to run the command without `cask`, then it definitely did not come from us, and you should find the source.

**Comment by adityadalal924** [comment]
Looks like the following stuff gets left behind with an uninstall. I'll update the TeamViewer Cask when I get a chance.

```
/Library/LaunchAgents/com.teamviewer.teamviewer_desktop.plist
/Library/LaunchAgents/com.teamviewer.teamviewer.plist
/Library/LaunchDaemons/com.teamviewer.Helper.plist
/Library/LaunchDaemons/com.teamviewer.teamviewer_service.plist
```

**Comment by guandalf** [comment]
Checked again on a fresh El Capitan installation. After:
brew cask install teamviewer
brew cask uninstall teamviewer
macos hang at boot (see attachment).

I cannot find the files @adityadalal924 mentioned and I checked without the fix.

![22779710039_843869c2eb_z](https://cloud.githubusercontent.com/assets/122050/11303117/822d2ad6-8fa1-11e5-9f66-1892bf75907b.jpg)

![22852059670_e0a85e53d3_z](https://cloud.githubusercontent.com/assets/122050/11303333/943c044e-8fa2-11e5-9cde-62e778cf0fb4.jpg)

**Comment by vitorgalvao** [comment]
Very weird. Well, since we could not reproduce, most we can do for now is reopen for a while and see if anyone else can.

**Comment by vitorgalvao** [comment]
In case you want to try again, we’d need some terminal output. From what I understand, it only hags at boot (i.e. you can still use your computer until you turn it off), correct? In that case, we’d need the outputs of something like:

```bash
brew doctor
brew cask doctor
brew install --verbose teamviewer
brew uninstall --verbose teamviewer
brew doctor
brew cask doctor
```

**Comment by adityadalal924** [comment]
The only other thing I can think of is seeing if using TeamViewer's inbuilt uninstall also causes the same error on your machine (We're not removing anything that this command doesn't) - Open the `TeamViewer` preferences > `Advanced` tab> Under `Uninstall`, click the `Uninstall` button

**Comment by guandalf** [comment]
@vitorgalvao unfortunately I have to fresh reinstall El Captain each time and this takes (lot) of time (old machine). I'm able to run the mac in single user mode and in repair mode, but neither of them are usable to test.

**Comment by SuperMarioBean** [comment]
hi everyone, I found this issues on my macbook pro, every time I run:

```
brew uninstall teamviewer
```

my macbook pro won't boot just like @guandalf mentioned
@adidalal use the teamviewer build-in `uninstall` can solve the problem
and I repeat this issue on my macbook air

there must be some difference between `brew cask uninstall teamview` and teamview build-in `uninstall`

I recommend there should display a warning for using `brew cask uninstall teamview` after teamview installation by brew cask

<img width="591" alt="image" src="https://user-images.githubusercontent.com/3890843/38475240-56dc7648-3bd9-11e8-81a3-2370577a4f4f.png">

**Comment by commitay** [comment]
@SuperMarioBean Can you post the output of `brew cask doctor` please?

**Comment by SuperMarioBean** [comment]
@commitay 
```
==> Homebrew-Cask Version
Homebrew-Cask 1.5.14-52-g775b632
caskroom/homebrew-cask (git revision bf586; last commit 2018-04-09)
==> macOS
10.13.4
==> SIP
Enabled
==> Java
10, 1.8.0_141
==> Homebrew-Cask Install Location
<NONE>
==> Homebrew-Cask Staging Location
/usr/local/Caskroom
==> Homebrew-Cask Cached Downloads
~/Library/Caches/Homebrew/Cask (1 files, 354MB)
==> Homebrew-Cask Taps:
/usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask (3955 casks)
/usr/local/Homebrew/Library/Taps/caskroom/homebrew-versions (173 casks)
==> Contents of $LOAD_PATH
/usr/local/Homebrew/Library/Homebrew/cask/lib
/usr/local/Homebrew/Library/Homebrew
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/gems/2.3.0/gems/did_you_mean-1.0.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/site_ruby/2.3.0
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/site_ruby/2.3.0/x86_64-darwin9.0
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/site_ruby/2.3.0/universal-darwin9.0
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/site_ruby
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/vendor_ruby/2.3.0
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/vendor_ruby/2.3.0/x86_64-darwin9.0
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/vendor_ruby/2.3.0/universal-darwin9.0
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/vendor_ruby
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/x86_64-darwin9.0
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/universal-darwin9.0
==> Environment Variables
LC_ALL="en_US.UTF-8"
PATH="/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/Homebrew/Library/Homebrew/shims/scm"
SHELL="/bin/zsh"
```

**Comment by commitay** [comment]
Can't reproduce.

**Comment by SuperMarioBean** [comment]
@commitay Did you open teamviewer.app and configure with guide after `brew cask install teamviewer`，then try `brew cask uninstall teamviewer`

WARNING: backup your data

**Comment by commitay** [comment]
> configure with guide

What does this mean?

**Comment by SuperMarioBean** [comment]
@commitay my apologize, `initial setup`
<img width="723" alt="image" src="https://user-images.githubusercontent.com/3890843/38476886-a03f0a72-3be2-11e8-9668-70ebbd69666a.png">

**Comment by commitay** [comment]
@SuperMarioBean The problem seems to be with uninstalling one of the pkgs. We could work around this but because upstream may make changes to the pkgs that might cause future issues we have instead completely removed the `uninstall`.

**Comment by SuperMarioBean** [comment]
@commitay a WARNING notes include this issues links in terminal output would be fine.
